rita_common/src/debt_keeper/mod.rs

```diff
@@ -606,6 +606,21 @@ impl DebtKeeper {
     }
 }
 
+/// On an interupt (SIGTERM), saving debtkeeper before exiting
+pub fn save_debt_on_shutdown() {
+    let mut dk = DEBT_DATA.write().unwrap();
+    let previous_save = dk.last_save;
+    dk.last_save = Some(Instant::now());
+
+    if let Err(e) = dk.save() {
+        error!("Failed to save debts {:?}", e);
+        dk.last_save = previous_save;
```

Comment:
(A) this can be an if/else and avoid the return statement, B what's up with the last save logic? I don't think it does anything give that the program is exiting